### PR TITLE
🎨🔢 Set limit to number of characters and possible options on question and list block options

### LINKED ISSUE #302: 🎨 🔢 As a user I should be able to see the number of characters in question and list block options
## User stories

As a user I want to only enter a specific number of characters in the question and list blocks so that I do not surpass the maximum characters allowed by API.

---

Quick links: [Summary](#summary) | [Description](#description) | [Template](#template) | [Example](#example) | [Resources](#resources)

---

### Description
 
The WhatsApp API allows:

1. For the list block

- A maximum of 10 options with a maximum of 24 characters each.

2. For the question block

- A maximum of 3 options with a maximum of 20 characters each

What is expected:

```markdown
1. A warning 'maximum characters reached' when characters reach the maximum.
2. A check to prevent creation of more options once the maximum number of options is reached.


### Acceptance Criteria

1. [If I pass 24 and 20 characters for the list and question block respectively]
1. [I should not be able to continue typing more characters.]

```

### PR DESCRIPTION
**Pull request by royokite**
# Description

This pull request is made in part to handle number of options and character limit on the question and list blocks.

For the list block:
- A maximum of 10 options with a maximum of 24 characters each.

For the question block
- A maximum of 3 options with a maximum of 20 characters each

Fixes #302 

## Type of change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)

# Screenshot (optional)
Screenshot (372)

# How Has This Been Tested?

- Type into option input and evaluate input size set to 20 for question block and 24 for list block.
- Created a question block to test possible options and the limit works at 3.
- Created a list block to test possible options and the limit works at 10.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: libs/features/convs-mgr/stories/blocks/library/block-options/src/lib/components/option-input-field/option-input-field.component.html [hunk 1]
```diff
@@ -4,7 +4,10 @@
       <div [formGroupName]="formGroupNameInput" class="option" fxFlex>
         <input [id]="inputUniqueId" [class]="optionClass" [readOnly]="isReadOnly"
         [ngClass]="isReadOnly? 'readonly' : ''" formControlName="message"
-          type="text" placeholder="Click here to edit" fxFlexFill>
+          type="text" placeholder="Click here to edit" [maxlength]="charMaxlength" [(ngModel)]="optionValue" fxFlexFill>
+        <span class="length-error" *ngIf="optionValue.length >= charMaxlength">
+          maximum characters reached
```
Comment (ReaganCn) [inline libs/features/convs-mgr/stories/blocks/library/block-options/src/lib/components/option-input-field/option-input-field.component.html:9]: Please use multi-lang. Have a look at the below from the `login.component.html`
image

The translations are added in the files `en.json` and `fr.json`

File: libs/features/convs-mgr/stories/blocks/library/list-message-block/src/lib/components/list-block/list-block.component.ts [hunk 2]
```diff
@@ -45,9 +45,9 @@ export class ListBlockComponent<T> implements OnInit, AfterViewInit {
   }
 
   addNewOption() {
-    this.listItems.push(this.addListOptions());
+    if (this.listItems.length < 10) this.listItems.push(this.addListOptions());
```
Comment (ReaganCn) [inline libs/features/convs-mgr/stories/blocks/library/list-message-block/src/lib/components/list-block/list-block.component.ts:48]: Save this number to a `const`

File: libs/features/convs-mgr/stories/blocks/library/question-message-block/src/lib/components/questions-block/questions-block.component.html [hunk 3]
```diff
@@ -10,7 +10,7 @@
         <div [formGroupName]="i" class="inputList" fxLayout="row" fxLayoutALign="start center" fxFlex="100">
           <button (click)="deleteInput(i)" class="listgone"><i class="far fa-trash-alt"></i></button>
           <app-option-input-field [jsPlumb]="jsPlumb" [blockFormGroup]="questionMessageBlock" [formGroupNameInput]="i"
-            class="input"></app-option-input-field>
+            class="input" [charMaxlength]=20></app-option-input-field>
```
Comment (ReaganCn) [inline libs/features/convs-mgr/stories/blocks/library/question-message-block/src/lib/components/questions-block/questions-block.component.html:13]: Same here

File: libs/features/convs-mgr/stories/blocks/library/question-message-block/src/lib/components/questions-block/questions-block.component.ts [hunk 4]
```diff
@@ -49,7 +49,7 @@ export class QuestionsBlockComponent implements OnInit, AfterViewInit {
   }
 
   addNewOption() {
-    this.options.push(this.addQuestionOptions());
+    if (this.options.length < 3) this.options.push(this.addQuestionOptions());
```
Comment (ReaganCn) [inline libs/features/convs-mgr/stories/blocks/library/question-message-block/src/lib/components/questions-block/questions-block.component.ts:52]: Same here

**Review by ReaganCn — CHANGES_REQUESTED** [review]
Good job. Just some minor changes

File: libs/features/convs-mgr/stories/blocks/library/question-message-block/src/lib/components/questions-block/questions-block.component.ts (hunk 4 above)
Comment (royokite) [reply libs/features/convs-mgr/stories/blocks/library/question-message-block/src/lib/components/questions-block/questions-block.component.ts]: Made all the requested changes. @ReaganCn

**Review by ReaganCn — APPROVED** [review]

**Pull request merged**